packages/core/src/actions/entries.ts

```diff
@@ -693,6 +694,30 @@ export function loadEntries(collection: Collection, page = 0) {
     const backend = currentBackend(configState.config);
 
     const loadAllEntries = 'nested' in collection || hasI18n(collection);
+    let loadPredicate = undefined;
+    if ('nested' in collection && collection.nested?.lazy_load && lazyLoadFolder !== null) {
+      const lazyLoadPath = [collection.folder, lazyLoadFolder].filter(p => !!p).join('/');
+      loadPredicate = (path: string) => {
+        // find length of shared part
+        const length = Math.min(lazyLoadPath.length, path.length);
+        let i = 0
+        for (; i < length && lazyLoadPath[i] === path[i]; ++i)
+          ;
+        const maxDepth = lazyLoadFolder && i == lazyLoadPath.length ? 2 : 1;
```

Comment:
Should we also have a minDepth? It doesn't make sense to me intuitively to load the whole tree below if the selection of a page triggers a reload, right?

I am thinking something like minDepth = maxDepth -1  (with some guard to not get negative)

Reply:
The changes are tailored to the needs of the `Sidebar` and "page detection" logic (ie. "what is an editable page in a Hugo page tree").

I'm pretty sure filtering more `Entry` downloads than the previously suggested changes would result in errors during rendering the tree.

Reply:
I see - it's strange that the whole page content is loaded to render the sidebar but let's go with this if further reduction is difficult right now.

Reply:
Yes, it's indeed a pretty crappy design. Many objects (whole files!) are loaded repeatedly without a smart way to reduce traffic.

As I mentioned before, the whole `NestedCollection` feature is more of an afterthought.

Reply:
A reason that may have contributed to these design choices is supporting many different backends/APIs.

Reply:
OK, let's go with this for now then.